app/memorial/controllers/display.py

```diff
@@ -48,3 +49,58 @@ def display_single_memorial_for_user(user_id, memorial_id):
         return "User does not have access to view the heirloom", 403
 
     return (memorial_doc.to_json(), role_doc.to_json() if role_doc != None else None), 200
+
+
+def display_members_for_single_memorial(user_id, memorial_id):
+    # Validate that the memorial Id exists
+    if not memorial_id:
+        return {"msg": "Missing memorialID parameter"}, 400
+
+    # Retrieves user_id role type for permissions method
+    role_id = RoleRepo.get_by_user_id(user_id) if user_id is not None else None
+
+    # Retrieves memorial used to fetch members
+    memorial_doc = MemorialRepo.get_by_id(memorial_id)
+
+    # Retrieves all roles that correspond to given memorial
+    role_docs = RoleRepo.get_by_memorial_id(memorial_id) if memorial_id is not None else None
+
+    # Validates that a role exist
+    if role_docs is None:
+        return "No role information found", 404
+
+    # Stores user_id for all roles connected to given memorial
+    user_ids = []
+    for role_doc in role_docs:
+        user_ids.append(str(role_doc.user))
+
+    # Retrieves all user documentation from user_ids provided
+    user_doc = UserRepo.get_users_by_ids(user_ids)
+
+    if user_doc is None:
+        return "No user information found", 404
+
+    if memorial_doc is None:
+        return "Memorial not found", 404
+
+    members = {}
+    user_roles = []
+
+    # Stores user role types for json display
+    for role_type in role_docs:
+        user_roles.append(str(role_type.role))
+
+    # Creates a JSON response displayed when api route is called
+    for mem in user_doc:
+        members["email"] = mem.email
+        members["firstName"] = mem.first_name
+        members["id"] = str(mem.id)
+        members["lastName"] = mem.last_name
+        members["role"] = user_roles.pop(0)
```

Comment:
![image](https://user-images.githubusercontent.com/31978033/110218284-b9487280-7e6d-11eb-90e9-9032c403166b.png)

i made these changes and it returned all of them

Reply:
locally, i didn't commit